### ISSUE TITLE
Orthogonal initialization fails for (at least) 2d matrices

### ISSUE THREAD

**Issue by ramasesh** [google/jax#1615]
The following code should generate an orthogonal 10x10 matrix.  

```
from jax.nn.initializers import orthogonal, uniform
from jax import random

key = random.PRNGKey(0)

o_init = orthogonal()
orthogonal_matrix = o_init(key, (10,10))
```

However, the actual output is the following:

```
--------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-af5241da1f40> in <module>
----> 1 o_init(key, (10,10))

~/VirtualEnvs/Jax/lib/python3.7/site-packages/jax/nn/initializers.py in init(key, shape, dtype)
     93     Q *= np.sign(np.diag(R)) # needed for a uniform distribution
     94     if n_rows < n_cols: Q = Q.T
---> 95     Q = np.reshape(Q, onp.delete(shape, column_axis) + (shape[column_axis],))
     96     Q = np.moveaxis(Q, -1, column_axis)
     97     return scale * Q

~/VirtualEnvs/Jax/lib/python3.7/site-packages/jax/numpy/lax_numpy.py in reshape(a, newshape, order)
    730 def reshape(a, newshape, order="C"):
    731   try:
--> 732     return a.reshape(newshape, order=order)  # forward to method for ndarrays
    733   except AttributeError:
    734     return _reshape(a, newshape, order=order)

~/VirtualEnvs/Jax/lib/python3.7/site-packages/jax/numpy/lax_numpy.py in _reshape_method(a, *newshape, **kwargs)
    760   if len(newshape) == 1 and not isinstance(newshape[0], int):
    761     newshape = newshape[0]
--> 762   return _reshape(a, newshape, order=order)
    763 
    764 

~/VirtualEnvs/Jax/lib/python3.7/site-packages/jax/numpy/lax_numpy.py in _reshape(a, newshape, order)
    736 def _reshape(a, newshape, order="C"):
    737   dummy_val = onp.broadcast_to(0, shape(a))  # zero strides
--> 738   computed_newshape = onp.reshape(dummy_val, newshape).shape
    739 
    740   if order == "C":

<__array_function__ internals> in reshape(*args, **kwargs)

~/VirtualEnvs/Jax/lib/python3.7/site-packages/numpy/core/fromnumeric.py in reshape(a, newshape, order)
    299            [5, 6]])
    300     """
--> 301     return _wrapfunc(a, 'reshape', newshape, order=order)
    302 
    303 

~/VirtualEnvs/Jax/lib/python3.7/site-packages/numpy/core/fromnumeric.py in _wrapfunc(obj, method, *args, **kwds)
     59 
     60     try:
---> 61         return bound(*args, **kwds)
     62     except TypeError:
     63         # A TypeError occurs if the object does have such a method in its

ValueError: cannot reshape array of size 100 into shape (20,)
```

As a sanity check, running almost the identical code for a uniform initialization works fine:
```
from jax.nn.initializers import orthogonal, uniform
from jax import random

key = random.PRNGKey(0)

u_init = uniform()
uniform_matrix = u_init(key, (10,10))
```

From looking at the code for the orthogonal initializer, it seems like the problem occurs after the QR decomposition is completed and the Q matrix is being reshaped.  Here is the source:
```
def orthogonal(scale=1.0, column_axis=-1):
   """
   Construct an initializer for uniformly distributed orthogonal matrices.
  
   If the shape is not square, the matrices will have orthonormal rows or columns
   depending on which side is smaller.
   """
   def init(key, shape, dtype=np.float32):
     if len(shape) < 2:
        raise ValueError("orthogonal initializer requires at least a 2D shape")
     n_rows, n_cols = onp.prod(shape) // shape[column_axis], shape[column_axis]
     matrix_shape = (n_cols, n_rows) if n_rows < n_cols else (n_rows, n_cols)
     A = random.normal(key, matrix_shape, dtype)
     Q, R = np.linalg.qr(A)
     Q *= np.sign(np.diag(R)) # needed for a uniform distribution
     if n_rows < n_cols: Q = Q.T
     Q = np.reshape(Q, onp.delete(shape, column_axis) + (shape[column_axis],))
     Q = np.moveaxis(Q, -1, column_axis)
     return scale * Q
    return init    
```

It looks as if the line ```Q = np.reshape(Q, onp.delete(shape, column_axis) + (shape[column_axis],))``` is trying to reshape the array into some shape, but that shape is not properly getting specified.  Specifically, the line ```onp.delete(shape, column_axis) + (shape[column_axis],)``` does not seem to be doing what it was intended to do.

**Comment by hawkinsp** [comment]
What version of the `jax` package do you have? I think this may be already fixed in the latest release (0.1.49).

**Comment by ramasesh** [comment]
Awesome, you are right.  I had (0.1.48).  I upgraded to (0.1.49) and the issue is fixed.  Thanks!
